hs/src/Reach/Connector/ALGO.hs

```diff
@@ -1300,6 +1311,16 @@ checkTxnInit vTypeEnum ct_mcsend = do
   -- [ txn ]
   return ()
 
+checkTxnUsage :: CheckTxn -> App ()
+checkTxnUsage (CheckTxn {..}) = do
+  case ct_mtok of
+    Just tok -> do
+      newToks <- (liftIO . readIORef) =<< asks eNewToks
+      when (tok `S.member` newToks) $ do
+            bad $ LT.pack $ "Token cannot be transferred within the same consensus step it was created in (at: " <> show ct_at <> ")"
```

Comment:
I know it is harder, but I think this needs to have an error code with a link to a longer explanation of how to fix it and why it is a problem. I don't know how reusable that stuff is, but sometimes we have to make our lives painful to help the users.

Reply:
If you don't agree with that, at the least it should say why: ", because on Algorand, you must opt-in to a token to receive it, but it is impossible to opt-in to a token unless you know its id, and the id cannot be known until after this transaction"